lib/widgets/message_list.dart

```diff
@@ -511,147 +534,202 @@ class _UnreadMarker extends StatelessWidget {
-              color: streamColor,
-              // TODO globe/lock icons for web-public and private streams
-              child: Text(streamName, style: TextStyle(color: contrastingColor)))),
-          Expanded(
-            child: Padding(
-              // Web has padding 9, 3, 3, 2 here; but 5px is the chevron.
-              padding: const EdgeInsets.fromLTRB(4, 3, 3, 2),
-              child: Text(topic,
-                // TODO: Give a way to see the whole topic (maybe a
-                //   long-press interaction?)
-                overflow: TextOverflow.ellipsis,
-                style: const TextStyle(fontWeight: FontWeight.w600)))),
-          // TODO topic links?
-          // Then web also has edit/resolve/mute buttons. Skip those for mobile.
-          RecipientHeaderDate(message: message),
-        ])));
+        color: backgroundColor,
+        child: Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            // TODO(#282): Long stream name will break layout; find a fix.
+            streamWidget,
+            Expanded(
+              child: Padding(
+                padding: const EdgeInsets.symmetric(vertical: 11),
+                child: Text(topic,
+                  // TODO: Give a way to see the whole topic (maybe a
+                  //   long-press interaction?)
+                  overflow: TextOverflow.ellipsis,
+                  style: textStyle))),
+            // TODO topic links?
+            // Then web also has edit/resolve/mute buttons. Skip those for mobile.
+            RecipientHeaderDate(message: message,
+              color: contrastingColor.withOpacity(0.4)),
+          ])));
   }
 }
 
-final _kStreamMessageBorderColor = const HSLColor.fromAHSL(1, 0, 0, 0.88).toColor();
-
 class DmRecipientHeader extends StatelessWidget {
   const DmRecipientHeader({super.key, required this.message});
 
   final DmMessage message;
 
   @override
   Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
     final store = PerAccountStoreWidget.of(context);
     final String title;
     if (message.allRecipientIds.length > 1) {
-      final otherNames = message.allRecipientIds
+      title = zulipLocalizations.messageListGroupYouAndOthers(message.allRecipientIds
         .where((id) => id != store.account.userId)
-        .map((id) => store.users[id]?.fullName ?? '(unknown user)')
+        .map((id) => store.users[id]?.fullName ?? zulipLocalizations.messageListGroupUnknownUser)
         .sorted()
-        .join(", ");
-      title = 'You and $otherNames';
+        .join(", "));
     } else {
-      title = 'You with yourself'; // TODO pick string; web has glitchy "You and $yourname"
+      // TODO pick string; web has glitchy "You and $yourname"
+      title = zulipLocalizations.messageListGroupYouWithYourself;
     }
 
     return GestureDetector(
-      behavior: HitTestBehavior.translucent,
```

Comment:
msglist: Remove HitTestBehavior.translucent on DmRecipientHeader
    
    This seems unintended as the header should capture all
    clicks within its bounds.

Capturing all clicks within its bounds is the reason this was `translucent`. Here's that value's doc:
```dart
  /// Translucent targets both receive events within their bounds and permit
  /// targets visually behind them to also receive events.
  translucent,
```

So with that value, the header does capture all taps within its bounds.

The default is `deferToChild`. That one's doc says:
```dart
  /// Targets that defer to their children receive events within their bounds
  /// only if one of their children is hit by the hit test.
  deferToChild,
```

So with that default, the header would *not* necessarily capture all taps within its bounds — it only does so if its child is hit.

This parameter is unnecessary now, though, so it's good to remove it. The reason it's unnecessary is that the child fills the whole box.

Looking at the history with `git log -L` (`git log -L 668,693:lib/widgets/message_list.dart` as of the tip of your branch, to cover this whole `return` statement), I believe the key change was 412233c9cdd87062bb63dba2a7d6da85cdddf440. Before that, and when this line was added in f7feabf773a02d4b34ada9956889d55e96f285f8 as part of making the DM recipient header touchable at all, the child was an `Align` and so it only occupied the left-hand portion of the parent. That made `HitTestBehavior.translucent` necessary in order for touching everywhere in the header to work.